mne_connectivity/datasets/frequency.py

```diff
@@ -0,0 +1,146 @@
+        Sampling frequency of the simulated data, in Hz.
+    trans_bandwidth : int | float (default 1)
+        Transition bandwidth of the filter to apply to isolate activity in
+        ``freq_band``, in Hz.
+    snr : float (default 0.7)
+        Signal-to-noise ratio of the simulated data in the range [0, 1].
+    connection_delay : int (default 5)
+        Number of timepoints for the delay of connectivity between the seeds and
+        targets. If > 0, the target data is a delayed form of the seed data. If < 0, the
+        seed data is a delayed form of the target data.
+    tmin : int | float (default 0)
+        Earliest time of each epoch.
+    ch_names : list of str | None (default None)
+        Names of the channels in the simulated data. If `None`, the channels are named
+        according to their index and the frequency band of interaction.
+    ch_types : str | list of str (default "eeg")
+        Types of the channels in the simulated data.
+    rng_seed : int | None (default None)
+        Seed to use for the random number generator. If `None`, no seed is specified.
+
+    Returns
+    -------
+    epochs : mne.EpochsArray
+        The simulated data stored in an `mne.EpochsArray` object. The channels are
+        arranged according to seeds, then targets.
+
+    Notes
+    -----
+    Signals are simulated as a single source of activity in the given frequency band and
+    projected into ``n_seeds + n_targets`` noise channels.
+    """
+    n_channels = n_seeds + n_targets
+
+    # check inputs
+    if n_seeds < 1 or n_targets < 1:
+        raise ValueError("Number of seeds and targets must each be at least 1.")
+
+    if not isinstance(freq_band, tuple):
+        raise TypeError("Frequency band must be a tuple.")
+    if len(freq_band) != 2:
+        raise ValueError("Frequency band must contain two numbers.")
+
+    if n_times < 1:
+        raise ValueError("Number of timepoints must be at least 1.")
+
+    if n_epochs < 1:
+        raise ValueError("Number of epochs must be at least 1.")
+
+    if sfreq <= 0:
+        raise ValueError("Sampling frequency must be > 0.")
+
+    if snr < 0 or snr > 1:
+        raise ValueError("Signal-to-noise ratio must be between 0 and 1.")
+
+    if np.abs(connection_delay) >= n_epochs * n_times:
+        raise ValueError(
+            "Connection delay must be less than the total number of timepoints."
+        )
+
+    # simulate data
+    rng = np.random.RandomState(rng_seed)
+
+    # simulate signal source at desired frequency band
+    signal = rng.randn(1, n_epochs * n_times + np.abs(connection_delay))
+    signal = filter_data(
+        data=signal,
+        sfreq=sfreq,
+        l_freq=freq_band[0],
+        h_freq=freq_band[1],
+        l_trans_bandwidth=trans_bandwidth,
+        h_trans_bandwidth=trans_bandwidth,
+        fir_design="firwin2",
+    )
+
+    # simulate noise for each channel
+    noise = rng.randn(n_channels, n_epochs * n_times + np.abs(connection_delay))
```

Comment:
```suggestion
    noise = rng.standard_normal(size=(n_channels, n_epochs * n_times + np.abs(connection_delay)))
```

https://numpy.org/doc/stable/reference/random/generated/numpy.random.randn.html

Reply:
Good to know!